### PR TITLE
[Bug] Fix show load like match

### LINKED ISSUE #6313: show load like match error
**Describe the bug**

When there are some matching records, the correct result is not returned

```
mysql>  show load where Label like '%a%';
Empty set (0.04 sec)
```

### PR DESCRIPTION
**Pull request by DinoZhang**
## Proposed changes

Matching should use patterns instead of contains

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #6313) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morningman** [comment]
This is a historical issue, because the previous `like` operation did not implement wildcards. If you change this way, all online operations using the previous method will fail.
Unless we can implement a compatible solution, such as automatically adding `%` symbols.
eg:
auto convert `like 'abc'` to `like "%abc%"`

**Comment by DinoZhang** [comment]
For historical tasks, the `%` symbol has been automatically added.

File: fe/fe-core/src/main/java/org/apache/doris/load/loadv2/LoadManager.java [hunk 1]
```diff
@@ -317,7 +320,13 @@ public void cancelLoadJob(CancelLoadStmt stmt, boolean isAccurateMatch) throws D
                 }
             } else {
                 for (Map.Entry<String, List<LoadJob>> entry : labelToLoadJobs.entrySet()) {
-                    if (entry.getKey().contains(stmt.getLabel())) {
+                    String label = stmt.getLabel();
+                    // Compatible with historical issues
+                    if (!label.contains("%")) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/load/loadv2/LoadManager.java:325]: This should be done in analyze() of CancelLoadStmt.

File: fe/fe-core/src/main/java/org/apache/doris/qe/DdlExecutor.java [hunk 2]
```diff
@@ -142,9 +142,15 @@ public static void execute(Catalog catalog, DdlStmt ddlStmt) throws Exception {
             }
         } else if (ddlStmt instanceof CancelLoadStmt) {
             boolean isAccurateMatch = ((CancelLoadStmt) ddlStmt).isAccurateMatch();
+            String label = ((CancelLoadStmt) ddlStmt).getLabel();
+            if (!isAccurateMatch && label != null) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/DdlExecutor.java:146]: This should be done in analyze() of CancelLoadStmt.

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java [hunk 3]
```diff
@@ -977,18 +977,25 @@ private void handleShowLoad() throws AnalysisException {
             ErrorReport.reportAnalysisException(ErrorCode.ERR_BAD_DB_ERROR, showStmt.getDbName());
         }
         long dbId = db.getId();
+        boolean isAccurateMatch = showStmt.isAccurateMatch();
+        String label = showStmt.getLabelValue();
+        if (!isAccurateMatch && label != null) {
+            if (!label.contains("%")) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java:983]: This should be done in analyze() of ShowLoadStmt.

File: fe/fe-core/src/main/java/org/apache/doris/load/Load.java [hunk 4]
```diff
@@ -1610,7 +1612,8 @@ public boolean isLabelExist(String dbName, String labelValue, boolean isAccurate
                 }
             } else {
                 for (Map.Entry<String, List<LoadJob>> entry : labelToLoadJobs.entrySet()) {
-                    if (entry.getKey().contains(labelValue)) {
+                    PatternMatcher matcher = PatternMatcher.createMysqlPattern(labelValue, CaseSensibility.LABEL.getCaseSensibility());
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/load/Load.java:1615]: `PatternMatcher.createMysqlPattern` can be put outside the `for` loop.
Same suggestion to other `PatternMatcher.createMysqlPattern`.

File: fe/fe-core/src/main/java/org/apache/doris/load/loadv2/LoadManager.java (hunk 1 above)
Comment (DinoZhang) [reply fe/fe-core/src/main/java/org/apache/doris/load/loadv2/LoadManager.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/qe/DdlExecutor.java (hunk 2 above)
Comment (DinoZhang) [reply fe/fe-core/src/main/java/org/apache/doris/qe/DdlExecutor.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java (hunk 3 above)
Comment (DinoZhang) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/load/Load.java (hunk 4 above)
Comment (DinoZhang) [reply fe/fe-core/src/main/java/org/apache/doris/load/Load.java]: +1

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**